doc/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,374 @@
+# Project migration guide for single API version operator
```

Comment:
Move this file to the `./website` tree. IMO, no need to keep a copy in the `./doc` tree since it will be gone soon.